### PR TITLE
Bug 1728827 - Expose Node.js entry point `@mozilla/glean/node`

### PR DESCRIPTION
**Pull request by brizental**
Just a bunch of placeholders for now. But the entry point works :) See: 

Screenshot 2021-09-02 at 19 09 36

### CODE REVIEW

File: glean/src/index/base.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    applicationId: string,
+    uploadEnabled?: boolean,
+    config?: ConfigurationInterface
+  ): Promise<void>
+} => {
+  return {
+    /**
+     * Initialize Glean. This method should only be called once, subsequent calls will be no-op.
+     *
+     * @param applicationId The application ID (will be sanitized during initialization).
```
Comment (chutten) [inline glean/src/index/base.ts:30]: sanitized how?

File: samples/node/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+import Glean from "@mozilla/glean/node";
+
+Glean.initialize("glean-sample-cli", true);
+console.log("Hey! Glean initialize succesfully!");
```
Comment (chutten) [inline samples/node/index.js:10]:
```suggestion
console.log("Hey! Glean initialized succesfully!");
```

Not sure that log-level logging will be enjoyed by all users. I'm guessing configurable logging will come later?

**Review by chutten — APPROVED** [review]
I'm still new to the codebase, but I think I feel comfortable enough approving these changes.

File: samples/node/index.js (hunk 2 above)
Comment (brizental) [reply samples/node/index.js:10]: This is inside the sample app, so our users will not see this log message. Also, it's just a placeholder until I implement a real sample app in [Bug 1728809](https://bugzilla.mozilla.org/show_bug.cgi?id=1728809).

File: glean/src/index/base.ts (hunk 1 above)
Comment (brizental) [reply glean/src/index/base.ts:30]:
```suggestion
     * @param applicationId The application ID (will be sanitized, if necessary).
```

**Pull request merged**